base/shell.jl

```diff
@@ -65,7 +65,7 @@ function shell_parse(str::AbstractString, interpolate::Bool=true;
                 j = k
             end
         elseif interpolate && !in_single_quotes && c == '$'
-            update_arg(s[i:j-1]); i = k; j = k
+            update_arg(s[i:prevind(s, j)]); i = k; j = k
```

Comment:
I don't understand why these changes are being made as a part of this PR.  What does this have to do with `SubString`?

If `j` is the index of `c`, then `j-1` is actually correct for `s::String` because `c=='$'` is ASCII, by the way.

Reply:
There are two reasons:
1. if `j` is the index of `c` if `c=='$'` then `j-1` does not have to be correct. Consider `s=s="∀\$"`, then `j=4` and `j-1` is an incorrect index (`s[j-1]` throws `UnicodeError` even under stable 0.6 before the changes; the changes only make it also throw an error when range indexing is used e.g. `s[i:j-1]`).
2. It is implemented in `SubString` PR as `s` is `SubString` because it is defined as `s = lstrip(str)` in line 13 and `lstrip` returns `SubString`.

Reply:
@stevengj Is this justification OK?

Reply:
Right, I was thinking of `j+1`, not `j-1`.